jscomp/ml/runtime_representation.ml

```diff
@@ -0,0 +1,242 @@
+let extract_concrete_typedecl :
+    (Env.t -> Types.type_expr -> Path.t * Path.t * Types.type_declaration) ref =
+  ref (Obj.magic ())
+
+type 'value value = Known of 'value | Unknown
+
+type object_property = {
+  key: string;
+  value: runtime_js_value list value;
+  optional: bool;
+}
+and runtime_js_value =
+  | String of {value: string value}
+  | Number of {value: string value}
+  | BigInt of {value: string value}
+  | Boolean of {value: bool value}
+  | NullLiteral
+  | UndefinedLiteral
+  | Array of {element_type: runtime_js_value value}
+  | Object of {
+      properties: object_property list;
+      can_have_unknown_properties: bool;
```

Comment:
not sure what this boolean does
the runtime representation is *never* exhaustive, you can always have more stuff
no code depends on the *absence* of properties right?

Reply:
open and closed objects have the same runtime representation right?